tests/acceptance/async_test_base.py

```diff
@@ -81,29 +100,38 @@ def start(self, adapter, ioloop_factory):
                                                        self.on_timeout)
             self._run_ioloop()
             self.assertFalse(self._timed_out)
+            self.assertIsNone(self._conn_open_error)
+            # Catch unexpected loss of connection
+            self.assertTrue(self._public_stop_requested,
```

Comment:
This fixes a nasty bug that would cause tests to pass when the connection establishment failed.